### PR TITLE
Replace neon by napi

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Replace neon by napi. With this we get better error handling, less and cleaner code.
Also might help in the future if we want to implement traits in bindings like this https://github.com/iotaledger/iota-sdk/commit/bf43fb3ea94016e8d71830bd8aae386ca2ec8624, because it's also easy to await on Promises from the JS side

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Running examples

### CODE REVIEW

**Review by Alex6323 — DISMISSED** [review]
looks nice!

File: bindings/nodejs/scripts/build.js
Comment (DaughterOfMars) [inline bindings/nodejs/scripts/build.js:1]: Is this still needed?

File: sdk/src/wallet/error.rs [hunk 1]
```diff
@@ -45,6 +45,9 @@ pub enum Error {
     /// Insufficient funds to send transaction.
     #[error("insufficient funds {available}/{required} available")]
     InsufficientFunds { available: u64, required: u64 },
+    /// Invalid event kind.
+    #[error("invalid event kind: {0}")]
+    InvalidEventType(u8),
```
Comment (thibault-martinez) [inline sdk/src/wallet/error.rs:52]: Under the `events` features?

File: sdk/src/wallet/error.rs [hunk 2]
```diff
@@ -45,6 +45,9 @@ pub enum Error {
     /// Insufficient funds to send transaction.
     #[error("insufficient funds {available}/{required} available")]
     InsufficientFunds { available: u64, required: u64 },
+    /// Invalid event kind.
+    #[error("invalid event kind: {0}")]
```
Comment (thibault-martinez) [inline sdk/src/wallet/error.rs]: Don't mix kind and type

File: bindings/nodejs/src/lib.rs [hunk 3]
```diff
@@ -1,70 +1,65 @@
...
-use tokio::runtime::Runtime;
+use napi::{Error, Result, Status};
+use napi_derive::napi;
 
-pub static RUNTIME: Lazy<Runtime> = Lazy::new(|| Runtime::new().unwrap());
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum NodejsError {
+    /// Bindings errors.
+    #[error("{0}")]
```
Comment (DaughterOfMars) [inline bindings/nodejs/src/lib.rs]:
```suggestion
    #[error(transparent)]
```

File: sdk/src/wallet/error.rs [hunk 4]
```diff
@@ -45,6 +45,9 @@ pub enum Error {
     /// Insufficient funds to send transaction.
     #[error("insufficient funds {available}/{required} available")]
     InsufficientFunds { available: u64, required: u64 },
+    /// Invalid event kind.
```
Comment (DaughterOfMars) [inline sdk/src/wallet/error.rs]: use type?

Comment (DaughterOfMars) [reply sdk/src/wallet/error.rs]: I was too late

File: bindings/nodejs/scripts/build.js
Comment (Thoralf-M) [reply bindings/nodejs/scripts/build.js:1]: I think so, but not 100% sure, don't have something with Windows ready to test

File: bindings/nodejs/src/lib.rs [hunk 5]
```diff
@@ -15,22 +15,22 @@ use napi_derive::napi;
 #[non_exhaustive]
 pub enum NodejsError {
     /// Bindings errors.
-    #[error("{0}")]
+    #[error("transparent")]
```
Comment (thibault-martinez) [inline bindings/nodejs/src/lib.rs]: With quotes 🤔 ?

Comment (Thoralf-M) [reply bindings/nodejs/src/lib.rs]: It compiled :trollface:

Comment (Thoralf-M) [reply bindings/nodejs/src/lib.rs]: Fixed

File: bindings/nodejs/src/lib.rs [hunk 6]
```diff
@@ -1,70 +1,65 @@
...
+use napi::{Error, Result, Status};
+use napi_derive::napi;
 
-pub static RUNTIME: Lazy<Runtime> = Lazy::new(|| Runtime::new().unwrap());
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum NodejsError {
+    /// Bindings errors.
+    #[error(transparent)]
+    Bindings(#[from] iota_sdk_bindings_core::Error),
```
Comment (thibault-martinez) [inline bindings/nodejs/src/lib.rs]: I think it's a bit weird as a variant name, knowing we're already inside a bindings module, maybe Core ?

Comment (Thoralf-M) [reply bindings/nodejs/src/lib.rs]: BindingsCore or just Core?

Comment (thibault-martinez) [reply bindings/nodejs/src/lib.rs]: I'd personally go with Core

Comment (Thoralf-M) [reply bindings/nodejs/src/lib.rs]: Renamed

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: bindings/nodejs/scripts/build.js
Comment (DaughterOfMars) [reply bindings/nodejs/scripts/build.js:1]: annoying

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**